.github/workflows/ci.yml

```diff
@@ -27,6 +27,10 @@ jobs:
           matrix:
             go-version: [1.15.x]
             os: [ubuntu-latest]
+            kubernetes:
+              - v1.18.15
+              - v1.19.7
+              - v1.20.2
```

Comment:
Im thinking whats the best approach to follow. We are fully compatible with a v1.18 from the go modules, but is not the case with a v1.19 neither a v1.20. In general this is very nice, we were missing this, but I think it would be better to have something like:
- v1.16.*
- v1.17.*
- v1.18* --> current

one we bump to v1.19, then it will look as:

- v1.17.*
- v1.18*
- v1.19* --> current

where we basically say, we currently support X version and we are backwards compatible with the last two previous ones. What do you think?

Reply:
I think the K8s support strategy is something we should discuss and document somewhere.

Regarding my changes in this PR, my idea was the following:

We were on Kube 1.18. Once a new Kube version comes out, we should open a PR to add that version to the test matrix to check if it works. Should that be the case, we can merge the PR. From that point in time on, we validate that further changes do not break the future Kubernetes version. It does not necessarily mean we officially support it, though. That's something we can make a more formal process.

Having said that, I can also add older versions to the test matrix and see what happens.

Is that okay?

Reply:
I have not much experience with Kube versions to be honest. From a purely programming language and coding background, I can understand to test the "next" thing already. Same logic kind of applies to Kube as well. We would like to know upcoming problems beforehand. On the other side, I would not like our tests to fail, because of a change in the next Kube versions. With that in mind, I would opt for testing current, plus one or maybe two older versions. More feedback on this is highly appreciated.

Reply:
With Kubernetes 1.16, the installation of Tekton fails:

```
Error from server (Invalid): error when creating "https://github.com/tektoncd/pipeline/releases/download/v0.20.1/release.yaml": ValidatingWebhookConfiguration.admissionregistration.k8s.io "validation.webhook.pipeline.tekton.dev" is invalid: webhooks[0].admissionReviewVersions: Invalid value: []string{"v1"}: must include at least one of v1beta1
```

So, I am just adding 1.17 to the matrix.

Regarding newer versions @HeavyWombat, as mentioned above, no magic would add that version number to the matrix when k8s makes a release. A human would have to add it and get it through a PR to prove that it is green. And once it is green, I would say we should not anymore introduce changes that break it (as we would have to fix them sooner or later anyway).

Reply:
ok, yes thanks for adding v1.17 and checking that v1.16 does not longer works for us. I still think v1.20 is too new, but if it passed the tests is fine to keep it.